lib/adagrams.rb

```diff
@@ -0,0 +1,128 @@
+def draw_letters
+  letter_pool_array = [
+    "A", "A", "A", "A", "A", "A", "A", "A", "A",
+    "B", "B",
+    "C", "C",
+    "D", "D", "D", "D",
+    "E", "E", "E", "E", "E", "E", "E", "E", "E", "E", "E", "E",
+    "F", "F",
+    "G", "G", "G",
+    "H", "H",
+    "I", "I", "I", "I", "I", "I", "I", "I", "I",
+    "J", "K",
+    "L", "L", "L", "L",
+    "M", "M",
+    "N", "N", "N", "N", "N", "N",
+    "O", "O", "O", "O", "O", "O", "O", "O",
+    "P", "P", "Q",
+    "R", "R", "R", "R", "R", "R",
+    "S", "S", "S", "S",
+    "T", "T", "T", "T", "T", "T",
+    "U", "U", "U", "U",
+    "V", "V",
+    "W", "W",
+    "X", "Y", "Y", "Z"
+  ]
+  drawn_letters = []
+  10.times do |letter|
+    drawn_letters << letter_pool_array.shuffle.pop
+  end
+  drawn_letters
+end
+
+def uses_available_letters? input, letter_bank
+  input_array = input.upcase.split('')
+  letter_hash = {}
+
+  letter_bank.each do |letter|
+    letter_hash[letter] = letter_hash[letter] ? letter_hash[letter] + 1 : 1
+  end
```

Comment:
This is a pretty tricky algorithm, especially this early in the course. That's not a bad thing, but there will probably be some groups that need a little extra help.

I do like the idea of exposing them to goofy hash tricks early when we release this code.